community/planning/grid_document.md

```diff
@@ -0,0 +1,808 @@
+support folder depth greater than one. Grid Document would use the common
+backward slash character, "/", to indicate nested folder references. The
+transaction to create a folder could take in a nested folder path, separated
+by "/". If one of the folders in the path does not exist, the transaction would
+be invalid. If all files in the path exist, the folder will be created within
+the destination folder. Furthermore, a file may also be saved within a nested
+folder. The `folder` argument in the `CreateFileAction` payload will also take
+a path separated by the backward slash character. If any of the folders in this
+path do not exist, the transaction is invalid.
+
+The full path of a folder in Grid Document state, including parent folders up to
+DocumentRoot, may be used to calculate state addresses rather than just the
+unique folder name. However, allowing more folders in state may also necessitate
+increasing the number of characters to refer to the folder in state addresses
+to avoid hash collisions. This may also necessitate more safe-guards to ensure
+state size does not grow exponentionally.
+
+#### Folder namespace prefixes
+
+Additionally, additional folder layers may be addressed in Grid Document by
+using the already existing folder address. This address can be further prefixed
+using 2-hex-characters, similar to the construction of the first part. The “00”
+prefix would be reserved for storing files. The formula for a file would be as
+follows:
+
+```
+“621dee” + “07” + “01” + Sha512(folder_name)[10:] + “00” + Sha512(file_name)[48:]
+```
+Grid Document may allow folders to be addressed under other folders using the
+following formula:
+
+```
+“621dee” + “07” + “01” + Sha512(folder_name)[10:] + “01” + Sha512(folder_name)[48:]
+```
+
+The remaining namespaces may be defined in the future, depending on the needs of
+the smart contract or specific implementations.
+
+### Access to Grid Document state
+
+Sharing documents securely is useful alongside other Grid smart contracts. A
+smart contract can have read or write access to folders and files using Grid
+Document’s addressing formula. This will be interesting in cases where users
+want to provide more context or corroborating inter company communications. A
+simple example of this would be a supplier uploading an image of the product
+in a purchase order as it is shipped off.
+
+### Permissions
+
+Grid Document does not initially enforce permissions, but may be extended to
+support both Grid Pike and Workflow. In both cases, Grid Document could validate
+permissions to access, create, or delete a file or folder. An organization would
+be indicated as the owner of a file or folder. The organization indicated as the
+owner defines if and how agents, within and outside of the organization, may
+interact with files and folders in state.
+
+#### Incorporating Pike
+
+Grid Document can validate permissions to access, create or delete a file or
```

Comment:
access is a read operation and is not controlled by the smart contract (mutation only).

Reply:
Will remove mentions of read/access perms